src/tcache.c

```diff
@@ -860,6 +860,25 @@ tsd_tcache_data_init(tsd_t *tsd, arena_t *arena) {
 	return false;
 }
 
+bool
+tsd_tcache_data_init_with_bin_settings(tsd_t *tsd, arena_t *arena,
+    cache_bin_info_t tcache_bin_info[TCACHE_NBINS_MAX]) {
+	assert(tcache_bin_info != NULL);
+	return tsd_tcache_data_init_impl(tsd, arena, tcache_bin_info);
+}
+
+/* Initialize auto tcache (embedded in TSD). */
+bool
+tsd_tcache_data_init(tsd_t *tsd, arena_t *arena) {
```

Comment:
Check if this can be made static too.

Reply:
Yeah let's move tcache_enabled_set here and make it extern instead of inline. As things kept growing, it's not a good inline function anymore. Generally we want to keep the interface tight and only expose when have to. This helps compiler to make better decisions and often text size too.